test/test_array_shape.jl

```diff
@@ -64,16 +66,82 @@ using ElasticArrays, ArraysOfArrays
         A = collect(reshape(1:6*6, 6, 6))
         ac1 = ValueAccessor(ArrayShape{Real}(2), 2)
         ac2 = ValueAccessor(ArrayShape{Real}(3), 3)
+
         @test @inferred(getindex(A, ac1, ac2)) == [21 27 33; 22 28 34]
         @test @inferred(view(A, ac1, ac2)) == [21 27 33; 22 28 34]
         @test @inferred(setindex!(A, [6 4 3; 2 1 5], ac1, ac2)) === A
         @test A[ac1, ac2] == [6 4 3; 2 1 5]
     end
 
+    let A = collect(reshape(1:5*5*5, 5, 5, 5))
+        ac1 = ValueAccessor(ArrayShape{Real}(size(A)[1]), 0)
+        ac2 = ValueAccessor(ArrayShape{Real}(1), 0)
+        ac3 = ValueAccessor(ArrayShape{Real}(1), 1)
+        ac4 = ValueAccessor(ArrayShape{Real}(1), 2)
+
+        @test @inferred(getindex(A, ac1, ac1, ac1)) == A
+        @test @inferred(getindex(A, ac2, ac2, ac2))[1] == A[1]
+        @test @inferred(getindex(A, ac4, ac4, ac4))[1] == 63
+
+        @test @inferred(view(A, ac1, ac1, ac1)) == A
+        @test @inferred(view(A, ac2, ac2, ac2))[1] == A[1]
+        @test @inferred(view(A, ac4, ac4, ac4))[1] == 63
+
+        first_layer = @inferred(getindex(A, ac1, ac1, ac2))
+        setindex!(A, first_layer, ac1, ac1, ac3)
```

Comment:
Note that to use setindex! with ValueAccessors:

```
# Replace the scalar type element of A[1] = 1 with A[2] = 2
v = getindex(A, ac2, ac2, ac2).+1

# With and Without ValueAccessor
setindex!(A, v, ac2, ac2, ac2)
setindex!(A, 2, 1, 1, 1)

# Doesn't work
setindex!(A, 2, ac2, ac2, ac2)
```

Reply:
These getindex calls should also be showing coverage on line 86 in array_shape.jl, but aren't.

Reply:
Hm, no idea why.